terraform/modules/network/main.tf

```diff
@@ -0,0 +1,161 @@
+################################################################################
+# VPC Definition
+################################################################################
+resource "aws_vpc" "this" {
+  cidr_block                       = var.vpc_cidr
+  assign_generated_ipv6_cidr_block = true
+  enable_dns_support               = var.enable_dns_support
+  enable_dns_hostnames             = var.enable_dns_hostnames
+
+  tags = {
+    Name = "eda-vpc"
+  }
+}
+
+################################################################################
+# Availability zones
+################################################################################
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+################################################################################
+# IG Definition
+################################################################################
+resource "aws_internet_gateway" "igw" {
+  vpc_id = aws_vpc.this.id
+  tags   = {
+    Name = "eda-ig"
+  }
+}
+
+################################################################################
+# Subnets Definition
+################################################################################
+
+resource "aws_subnet" "public" {
+  count = var.public_subnet_count
+
+  vpc_id            = aws_vpc.this.id
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+
+  cidr_block      = cidrsubnet(aws_vpc.this.cidr_block, 4, count.index)
+  ipv6_cidr_block = cidrsubnet(aws_vpc.this.ipv6_cidr_block, 8, count.index)
+
```

Comment:
why do we remove this prop? `map_public_ip_on_launch`

map_public_ip_on_launch - (Optional) Specify true to indicate that instances launched into the subnet should be assigned a public IP address. Default is false.

Reply:
I think we don't need default public ip for our instances. We are only accessing them through the ELB